ideascube/serveradmin/views.py

```diff
@@ -111,3 +112,47 @@ def backup(request):
 def battery(request):
     return render(request, 'serveradmin/battery.html',
                   {'batteries': batinfo.batteries()})
+
+
+@staff_member_required
+def wifi(request, ssid=''):
+    wifi_list = AvailableWifiNetwork.all()
+
+    if ssid:
+        for wifi in wifi_list:
+            if wifi.ssid == ssid:
+                wifi_key = request.POST.get('wifi_key', '')
+
+                try:
+                    wifi.connect(wifi_key=wifi_key)
+                    messages.success(
+                        request, _("Connected to {ssid}").format(ssid=ssid))
+
+                except Exception as e:
+                    messages.error(request, e)
+
+                break
+
+    return render(request, 'serveradmin/wifi.html', {'wifi_list': wifi_list})
+
+
+@staff_member_required
+def wifi_history(request):
+    wifi_list = KnownWifiConnection.all()
+
+    if request.method == 'GET':
```

Comment:
I'd do the opposite: check if we are in POST to process request.POST, OR in any other case respond as if we are in GET mode. That's stronger (if you have a HEAD or OPTIONS request, you'll in trouble, with the current implementation ;) ).